docs/deploy.md

```diff
@@ -0,0 +1,66 @@
+# Deploying Cluster API Provider Openshift BareMetal
```

Comment:
I think this should probably be a section of the main README.md

Reply:
Moved it, thanks!